dart/lib/src/error_cause_extractor.dart

```
@@ -0,0 +1,21 @@
+class CauseExtractor<T> {
+  Object? cause(T error) { return null; }
```

This should return the error and the stack trace, but I guess you left that out for the sake of the example. 

Unfortunately, because exceptions don't have their stacktrace as a property, that can cause all kinds of weird behavior, like the following combinations:

- An exception has an inner exception plus the inner stacktrace
- An exception has just an inner stacktrace
- An exception has just an inner exception
- An exception has neither (but that's the same as no extraction, I guess)

---

Yeah left it out, just a jumping off point here. I also need to check for circles in the recursion. Not sure if this is realistic, but the Java SDK also does this and it's not much effort to do.